packages/tools/devtools/devtools-browser-extension/src/popup/PopupView.tsx

```diff
@@ -0,0 +1,108 @@
+	GetDevtoolsFeatures,
+	type ISourcedDevtoolsMessage,
+	type InboundHandlers,
+	handleIncomingMessage,
+} from "@fluidframework/devtools-core";
+import { type BackgroundConnection } from "../BackgroundConnection";
+
+const queryTimeoutInMilliseconds = 2000; // 2 seconds
+
+const loggingContext = "FLUID_DEVTOOLS(Popup)";
+
+/**
+ * Message sent to the webpage to query for the supported set of Devtools features.
+ */
+const getSupportedFeaturesMessage = GetDevtoolsFeatures.createMessage();
+
+/**
+ * Props containing critical BG service connection used for message passing with webpage
+ */
+export interface PopupViewProps {
+	backgroundServiceConnection: BackgroundConnection;
+}
+
+/**
+ * Component that renders when the user clicks the extension icon *in the browser toolbar*
+ * @returns popup component
+ */
+export function PopupView(props: PopupViewProps): React.ReactElement {
+	const { backgroundServiceConnection } = props;
+
+	// Indicates if Fluid Devtools is running in the current page (i.e. the application is using it and initialized it correctly).
+	// Undefined means still looking or that the message which is sent to discover this has not been sent yet.
+	const [foundDevtools, setFoundDevtools] = React.useState<boolean | undefined>(undefined);
+
+	React.useEffect(() => {
+		/**
+		 * Handlers for inbound messages.
+		 */
+		const inboundMessageHandlers: InboundHandlers = {
+			[DevtoolsFeatures.MessageType]: async (untypedMessage) => {
+				// We don't actually care what the supported features are here, we're just using the response
+				// as verification that Fluid Devtools are running on the page.
+				setFoundDevtools(true);
+				clearTimeout(responseTimeout);
+				return true;
+			},
+		};
+
+		/**
+		 * Event handler for messages coming from the Background Script.
+		 */
+		function messageHandler(message: Partial<ISourcedDevtoolsMessage>): void {
+			handleIncomingMessage(message, inboundMessageHandlers, {
+				context: loggingContext,
+			});
+		}
+		backgroundServiceConnection.on("message", messageHandler);
+
+		// Query for supported feature set
+		backgroundServiceConnection.postMessage(getSupportedFeaturesMessage);
+		let responseTimeout: NodeJS.Timeout | undefined;
+		if (foundDevtools === undefined) {
+			responseTimeout = setTimeout(() => {
+				setFoundDevtools(false);
+			}, queryTimeoutInMilliseconds);
+		}
+
+		return (): void => {
+			backgroundServiceConnection.off("message", messageHandler);
+		};
+	}, [backgroundServiceConnection, setFoundDevtools]);
+
+	// TODO: spinner for loading
+	// TODO: retry button on not found.
+	return (
+		<div>
+			{foundDevtools === undefined && (
+				<div>Searching for Fluid Devtools in the current tab...</div>
+			)}
+			{foundDevtools === true && (
+				<div>
+					Devtools found! Open the browser`&apos;`s devtools panel to view the Fluid
+					Devtools extension.
+				</div>
+			)}
+			{foundDevtools === false && (
+				<div>
+					Devtools not found. Visit the documentation{" "}
```

Comment:
I would vote for going into more detail here. E.g. "Fluid Devtools were not found running in the current tab. For details on how to enable the Devtools in your Fluid application, please refer to our documentation..."

@kashms interested in your thoughts on exact messaging for these. / If we should include any graphics or anything.

Reply:
+1 for adding more details here. "Fluid Devtools" can be confused for both the extension and/or the client library package so we should be more precise that it's about the latter.

`Fluid Devtools library was not found running in your Fluid application in the current tab. For details on how to enable it, please refer to our documentation here`